### PR TITLE
essence

### LINKED ISSUE #314: Essence Display under catacombs
Make it so that under the dungeon tab we can see how much essence the player has collected. Like say under dungeons, above Catacombs, there would be a list and the amount next to the name

- Ice Essence: 13 
- Wither Essence: 635

### PR DESCRIPTION
**Pull request by metalcupcake5**
dungeon essence display
firefox_ucqvS4lSiW
closes #314

### CODE REVIEW

**Comment by dukio** [comment]
honestly I think there's too much hardcoded stuff in lib.js and stats.ejs... it should be in constants 🤔 

if I manage to make SkyCrypt work in local again I'll try to make a PR

**Comment by metalcupcake5** [comment]
what would you move?

**Comment by dukio** [comment]
> what would you move?

https://github.com/SkyCryptWebsite/SkyCrypt/pull/1053

File: views/stats.ejs [hunk 1]
```diff
@@ -2128,6 +2128,17 @@ const metaDescription = getMetaDescription()
...
+            <div class="floor-icon" style="background-image: url(/head/33ff416aa8bec1665b92701fbe68a4effff3d06ed9147454fa77712dd6079b33)"></div> <span class="stat-name">Dragon: </span><span class="stat-value"><%= calculated.dungeons.essence.dragon %></span><br>
+            <div class="floor-icon" style="background-image: url(/head/8816606260779b23ed15f87c56c932240db745f86f683d1f4deb83a4a125fa7b)"></div> <span class="stat-name">Gold: </span><span class="stat-value"><%= calculated.dungeons.essence.gold %></span><br>
+            <div class="floor-icon" style="background-image: url(/head/964f25cfff754f287a9838d8efe03998073c22df7a9d3025c425e3ed7ff52c20)"></div> <span class="stat-name">Diamond: </span><span class="stat-value"><%= calculated.dungeons.essence.diamond %></span><br>
+            <div class="floor-icon" style="background-image: url(/head/ddba642efffa13ec3730eafc5914ab68115c1f998803f74452e2e0cd26af0b8)"></div> <span class="stat-name">Ice: </span><span class="stat-value"><%= calculated.dungeons.essence.ice %></span><br>
+          </p>
+      </div>
```
Comment (dukio) [inline views/stats.ejs:2141]: this last </div> shouldn't be here btw

Comment (nstringham) [reply views/stats.ejs:2141]: wdym?

**Review by dukio — APPROVED** [review]
lgtm, tested

preview (the one in the first PR post is outdated)
image

File: views/stats.ejs (hunk 1 above)
Comment (dukio) [reply views/stats.ejs]: Woops, meant the last `</div>` was extra, anyway fixed in my refactor (merged in this branch already)

**Comment by dukio** [comment]
@nstringham can you have a look at this, the `package-lock.json` is now using `lockfileVersion: 2` and idk if we should keep it or revert to `lockfileVersion 1`

**Comment by nstringham** [comment]
> idk if we should keep it or revert to `lockfileVersion 1`

that probably depends on what order things get merged in

**Comment by dukio** [comment]
> > idk if we should keep it or revert to `lockfileVersion 1`
> 
> that probably depends on what order things get merged in

you mean if this is merged after node v16, npm v7+ it should be fine right?

**Review by MartinNemi03 — APPROVED** [review]

**Pull request merged**